### PR TITLE
Unify backend-code for counters in list views

### PR DESCRIPTION
**Pull request by chrisvanrun**
Counts of found/filtered results for Archives/Reader Studies/Challenges/Algorithms are now displayed using the same backend logic.

For instance:
- "_No algorithms found._"
- "_1 algorithm found_"
- "_10 algorithms found_"

Or when a filter was applied:
- "_Filters included 1 of 10 total algorithms_"

Addresses #1659

### CODE REVIEW

**Review by MikeOverkamp-diag — APPROVED** [review]
LGTM!

File: app/grandchallenge/algorithms/views.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def form_valid(self, form):
         return response
 
 
-class AlgorithmList(PermissionListMixin, FilterMixin, ListView):
+class AlgorithmList(FilterMixin, PermissionListMixin, ListView):
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/views.py:88]: Why have the mixins changed order here? Is this to get the number of objects pre-permissions filtering?

Comment (chrisvanrun) [reply app/grandchallenge/algorithms/views.py:88]: The other way around. It's to get the correct number of objects on the context generation in the `FilterMixin`; post-permissions. The method resolution order had the permissions exclude objects after the `FilterMixin` making the mixin count objects that would not have been visible to the user anyway.

Comment (jmsmkn) [reply app/grandchallenge/algorithms/views.py:88]: You're right, my mistake!

**Pull request merged**